### ISSUE TITLE
Problem with categorial datasets in base-classifier from sklearn

### ISSUE THREAD

**Issue by darosantos** [darosantos/RFNS#2]
The project uses sklearn Decision Trees as its base classifier, but they are limited to working with numeric data, so how to handle code for categorical datasets.

**Comment by darosantos** [comment]
One way to solve the problem is to include preprocessing routines in the code.

**Comment by darosantos** [comment]
Another way, but discarded for now, is to implement the Decision Tree algorithm.

**Comment by darosantos** [comment]
Preprocessing routines should basically consist of turning categorical attributes into numeric ones .... For this One Hot Encoder may be the most interesting... https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html#sklearn.preprocessing.OneHotEncoder

**Comment by darosantos** [comment]
But the One Hot Enconder technique implemented in sklearn generates a new problem, if the column with numeric values is recognized as categorical then we will encode each value of this ... well, in the worst case we will have in the dataframe a number of columns equal to number of rows added to the rest of the other coded columns .... One way to solve this would be as indicated by the sklearn documentation and to use ColumnTransformer ... https://scikit-learn.org/stable/modules/generated/sklearn .compose.ColumnTransformer.html

**Comment by darosantos** [comment]
But using ColumnTransformer may not be so good because the code gets more complex and specific ... ColumnTransformer https://scikit-learn.org/stable/auto_examples/compose/plot_column_transformer_mixed_types.html is better suited for a pre pipeline -processing .... the code would end up being specific to some datasets and complex to end up trying to draw a pipeline that fits those dataset

**Comment by darosantos** [comment]
After some testing the most appropriate solution for the current version of the project is to loop the dataframe and go using OneHotEncode for columns that have possibly categorical data types (any non-numeric type)

**Comment by darosantos** [comment]
The solution adopted to deal with categorical data was to implement in code the use of OneHotEncoder applied to categorical columns, LabelEncoder for the target class and StandardScaler for normalization of numeric columns.

**Comment by darosantos** [comment]
> The solution adopted to deal with categorical data was to implement in code the use of OneHotEncoder applied to categorical columns, LabelEncoder for the target class and StandardScaler for normalization of numeric columns.

Included is the possibility for the user to configure which method will be used to encode categorical data, target class data and normalization.

**Comment by darosantos** [comment]
> > The solution adopted to deal with categorical data was to implement in code the use of OneHotEncoder applied to categorical columns, LabelEncoder for the target class and StandardScaler for normalization of numeric columns.
> 
> Included is the possibility for the user to configure which method will be used to encode categorical data, target class data and normalization.

However, normalization regarding data coding must be explicitly triggered by the marking attributes and especially before the fit method.
